core/citrus-spring/src/main/java/com/consol/citrus/common/XmlTestLoader.java

```diff
@@ -91,7 +98,25 @@ private ApplicationContext loadApplicationContext() {
                     new String[]{
                             getSource(),
                             "com/consol/citrus/spring/annotation-config-ctx.xml"},
-                    true, getParentApplicationContext());
+                    true, getParentApplicationContext()) {
+
+                /**
+                 * Configures the namespace handler resolver with the {@link XmlTestLoaderConfiguration}
+                 * @param reader
+                 */
+                @Override
+                protected void initBeanDefinitionReader(XmlBeanDefinitionReader reader) {
```

Comment:
id probably create a `CitrusClassPathXmlApplicationContext` just for the sake of readability..

Reply:
This is not possible as loaderConfig is referenced in the overloaded initBeanDefinitionReader() AND initBeanDefinitionReader() is called from the constructor of the application context. Thus there is no simple way to pass in the required loaderConfig - at least I don't see an other solution than creation of anonymous class here.